xclim/sdba/adjustment.py

```diff
@@ -1204,6 +1208,80 @@ def _adjust(
         return out
 
 
+class OTC:
+    """OTC"""
+
+    @classmethod
+    def _adjust(
+        cls,
+        ref,
+        hist,
+        bin_width=None,
+        bin_origin=None,
+        numItermax=100_000_000,
+    ):
+
+        out = xr.apply_ufunc(
+            otc_adjust,
+            ref,
+            hist,
+            kwargs=dict(
+                bin_width=bin_width,
+                bin_origin=bin_origin,
+                numItermax=numItermax,
+            ),
+            join="outer",
+            input_core_dims=[["time", "multivar"], ["time", "multivar"]],
+            output_core_dims=[["time", "multivar"]],
+            keep_attrs=True,
+            vectorize=True,
+        )
+
+        return out.T
+
+
+class dOTC:
+    """dOTC"""
+
+    @classmethod
+    def _adjust(
+        cls,
+        ref,
+        hist,
+        sim,
+        bin_width=None,
+        bin_origin=None,
+        cov_factor="std",
+    ):
+
+        ref = ref.rename(time="time_cal")
+        hist = hist.rename(time="time_cal")
+        sim = sim.rename(time="time_tgt")
+
+        out = xr.apply_ufunc(
+            dotc_adjust,
+            ref,
+            hist,
+            sim,
+            kwargs=dict(
+                bin_width=bin_width,
+                bin_origin=bin_origin,
+                cov_factor=cov_factor,
+            ),
+            join="outer",
+            input_core_dims=[
+                ["time_cal", "multivar"],
+                ["time_cal", "multivar"],
+                ["time_tgt", "multivar"],
+            ],
+            output_core_dims=[["time_tgt", "multivar"]],
+            keep_attrs=True,
```

Comment:
Which `attrs` are kept?  We should keep those of `sim`. Same remark with `hist` for OTC

Reply:
Thank you for taking a look @coxipi ! I believe this should be resolved now.

Reply:
Good! It makes think, there should really be information about both. Information about the original source (`hist, sim`  is the most crucial, but if we're talking about a scenario, we should also be informed about the target. This is not a point for your specific work here, just a thought I'm having, I will check how we handle this.